### ISSUE TITLE
`possibly-undefined` false positive with immutable bools

### ISSUE THREAD

**Issue by lucascolley** [KotlinIsland/basedmypy#826]
### Describe the problem

```python
def func(x: bool) -> None:
    if not x:
        msg = 'hello world'
    print("hi")
    if x:
        return
    print(msg)
```

This is type-safe. But:

`src/array_api_extra/_utils.py:71:11: error: Name "msg" may be undefined  [possibly-undefined]`

Related pylint issue: https://github.com/pylint-dev/pylint/issues/9662

### Severity

annoying but workaround is available

### Your Environment

basedmypy 2.7.0 (compiled: no)
Based on mypy 1.13.0

**Comment by KotlinIsland** [comment]
duplicate of #644

thanks for the report, this one is not so simple to resolve. the workaround is fairly straightforward, just assign a placeholder value beforehand

but if it's important to you I can devote some time to it 🙂

**Comment by lucascolley** [comment]
not high priority I suppose :)
